src/coreclr/jit/codegenarm.cpp

```diff
@@ -1120,7 +1120,16 @@ void CodeGen::genCodeForStoreLclVar(GenTreeLclVar* tree)
             else // store into register (i.e move into register)
             {
                 // Assign into targetReg when dataReg (from op1) is not the same register
-                inst_Mov(targetType, targetReg, dataReg, /* canSkip */ true);
+                if (!data->isUsedFromReg())
```

Comment:
When is this branch taken?

Reply:
From the xarch side of things, this is the comment:
```cpp
            if (!op1->isUsedFromReg())
            {
                // Currently, we assume that the non-reg source of a GT_STORE_LCL_VAR writing to a register
                // must be a constant. However, in the future we might want to support an operand used from
                // memory.  This is a bit tricky because we have to decide it can be used from memory before
                // register allocation,
                // and this would be a case where, once that's done, we need to mark that node as always
                // requiring a register - which we always assume now anyway, but once we "optimize" that
                // we'll have to take cases like this into account.
                assert((op1->GetRegNum() == REG_NA) && op1->OperIsConst());
                genSetRegToConst(targetReg, targetType, op1);
            }
            else
            {
                assert(targetReg == lclNode->GetRegNum());
                assert(op1->GetRegNum() != REG_NA);
                inst_Mov_Extend(targetType, /* srcInReg */ true, targetReg, op1->GetRegNum(), /* canSkip */ true,
                                emitTypeSize(targetType));
            }
```

Off the top of my head, I don't know what code causes to go to that path.

Reply:
Well, I just tried and the assertion gets hit :) 
>   Assertion failed 'data->OperIsConst()' in 'System.Enum:ToObject(System.Type,System.Object):System.Object' during 'Gen
  erate code' (IL size 355; hash 0x13b8126d; FullOpts)

Reply:
`!isUsedFromReg()` would imply here that the source is contained (or used from a spill temp).

The former implementation did not expect any contained operands, so I do not see the need for this branch.

Reply:
Well, I can't simply remove the branch and always call `inst_Mov_Extend` as I get a lot of asserts.

Reply:
That goes back to my original question then: when is `!isUsedFromReg()` `true` here?

Reply:
Trying to only use `inst_Mov_Extend` but the assertions are because of vector and zero registers. So, instead of using `!isUsedFromReg` for the branch, I will check to see if we are using a general register.

Reply:
Ok, for ARM32 it looks like we should be able to get rid of the branch, but not for ARM64.